TensorLib/Basic.lean

```diff
@@ -2,4 +2,98 @@ namespace TensorLib
 
 def hello := "world"
 
+class TensorElement (a: Type) where
```

Comment:
You may consider use the variable mechanism for a type like a, and also using {} to have it inferred. However, in this case, it may be good to start like this.

Reply:
I see what you mean. I'll be explicit/pedantic for now.